agent/consul/state/catalog.go

```diff
@@ -164,6 +207,7 @@ func init() {
 	registerSchema(servicesTableSchema)
 	registerSchema(checksTableSchema)
 	registerSchema(gatewayServicesTableNameSchema)
+	registerSchema(topologyTableNameSchema)
```

Comment:
What populates the new table with old data that predates the introduction of the hooks you added. You may have to shim in something into the snapshot restore logic.

Reply:
That should be taken care of since the topology table update happens on every `ensureServiceTxn`.

`Restore.Registration` -> `s.ensureRegistrationTxn` -> `ensureServiceTxn` -> `updateMeshTopology`

Reply:
:+1: